misc/python/materialize/cli/lint-cargo.py

```diff
@@ -0,0 +1,57 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+"""Check our set of Cargo.toml files for issues"""
+
+import sys
+from pathlib import Path
+from pprint import pprint
+from typing import Dict, List, Optional
+
+from materialize.cargo import Crate, Workspace
+
+
+def check_rust_versions() -> bool:
+    """Checks that every crate has a minimum specified rust version, and furthermore,
+    that they are all the same."""
+
+    def is_relevant(crate: Crate) -> bool:
+        # The `play` directory is for
+        # random miscellaneous stuff that isn't part of
+        # the Materialize product proper; as such, we shouldn't
+        # lint it.
+        return not crate.path.parts[-2] == "play"
```

Comment:
We currently hold `play` to the same linting standards as the rest of the codebase. Among other things `bin/check` still applies, so probably makes sense to force the same Rust version there. Were there particular problems here? We're an inch away from getting rid of the `play` directory entirely (see https://github.com/MaterializeInc/developer-experience/issues/1), so slight preference for avoiding special casing it since we'll have to remember to remove the special case in a week or two.